### ISSUE TITLE
Type issue with `QueryStringAddon` and `resolve`

### ISSUE THREAD

**Issue by jsardev** [elbywan/wretch#222]
I've been trying to create a reusable client with automatic expired token refresh.

Unfortunately, when adding the `QueryStringAddon` something's getting broken with the typings and I can't create a global `resolve` with `unauthorized`.

I'm getting this TS error:
```
The 'this' context of type 'Wretch<QueryStringAddon, unknown, undefined>' is not assignable to method's 'this' of type 'QueryStringAddon & Wretch<QueryStringAddon, unknown, undefined>'.
  Property 'query' is missing in type 'Wretch<QueryStringAddon, unknown, undefined>' but required in type 'QueryStringAddon'.(2684)
```

Here's a reproduction: https://stackblitz.com/edit/vitejs-vite-5per8h?file=src%2Fapi.ts

Would appreciate any help 🙏

**Comment by jsardev** [comment]
That was fast 😅 Thank you @elbywan 🙇‍♂️

**Comment by elbywan** [comment]
Hey @jsardev,

I just published `v2.8.1` which should now work as expected 🤞.


###### The `request` type should now be correct:
<img width="813" alt="Capture d’écran 2024-03-07 à 08 32 32" src="https://github.com/elbywan/wretch/assets/3428394/7caa66a6-abf1-414f-b197-bcdbf662bbbf">
